server/src/com/cloud/user/AccountManagerImpl.java

```diff
@@ -2177,6 +2173,7 @@ private UserAccount getUserAccount(String username, String password, Long domain
             if (domain != null) {
                 domainName = domain.getName();
             }
+            userAccount = _userAccountDao.getUserAccount(username, domainId);
```

Comment:
For now this is not an issue as every authenticator checks for the presence of user account. But the code flow of getUserAccount() method doesn't clearly brings it up. Earlier since there was a check for user account at the beginning it was very explicit. Now there is an implicit assumption that authenticator should return failure if the user account is not present in CCP. Please file a tracking bug for this.